tests/foreman/ui/test_products.py

```diff
@@ -220,13 +207,18 @@ def test_remove_prd(self, prd_name):
 
         """
         with Session(self.browser) as session:
-            make_product(
-                session,
-                org=self.organization.name,
-                loc=self.loc.name,
-                name=prd_name,
-                description=gen_string('alphanumeric'),
-            )
-            self.assertIsNotNone(self.products.search(prd_name))
-            self.products.delete(prd_name)
-            self.assertIsNone(self.products.search(prd_name))
+            for prd_name in generate_strings_list():
+                with self.subTest(prd_name):
+                    make_product(
+                        session,
+                        org=self.organization.name,
+                        loc=self.loc.name,
+                        name=prd_name,
+                        description=gen_string('alphanumeric'),
+                    )
+                    self.assertIsNotNone(self.products.search(prd_name))
+                    self.products.delete(prd_name)
+                    # Note: refresh is used here because sometimes selenium
+                    # is too fast to check the deleted object and it fails
+                    self.browser.refresh()
```

Comment:
I suspect it's because of custom `products.delete()` implementation. AFAIK @oshtaier implemented some generic `delete_entity()` method, which deletes entity and waits while it actually gets removed from UI.
The solution here is not to find better workaround, but to remove the need in workaround, i.e. remove the product correctly so there won't be need in any refresh or waiter. Although I'm afraid it may be out of the scope of this PR.